osu.Game/Overlays/News/Sidebar/MonthPanel.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics;
+using osu.Game.Online.API.Requests.Responses;
+using osu.Game.Graphics.Containers;
+using osuTK;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Graphics;
+using System.Linq;
+using System.Collections.Generic;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics.Sprites;
+
+namespace osu.Game.Overlays.News.Sidebar
+{
+    public class MonthPanel : CompositeDrawable
+    {
+        public readonly BindableBool IsOpen = new BindableBool();
+
+        private readonly FillFlowContainer postsFlow;
+
+        public MonthPanel(List<APINewsPost> posts)
+        {
+            Width = 160;
+            AutoSizeDuration = 250;
+            AutoSizeEasing = Easing.OutQuint;
+            InternalChild = new FillFlowContainer
+            {
+                RelativeSizeAxes = Axes.X,
+                AutoSizeAxes = Axes.Y,
+                Direction = FillDirection.Vertical,
+                Spacing = new Vector2(0, 5),
+                Children = new Drawable[]
+                {
+                    new DropdownButton(posts[0].PublishedAt)
+                    {
+                        IsOpen = { BindTarget = IsOpen }
+                    },
+                    postsFlow = new FillFlowContainer
+                    {
+                        RelativeSizeAxes = Axes.X,
+                        AutoSizeAxes = Axes.Y,
+                        Direction = FillDirection.Vertical,
+                        Spacing = new Vector2(0, 5),
+                        Children = posts.Select(p => new PostButton(p)).ToArray()
+                    }
+                }
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            IsOpen.BindValueChanged(open =>
+            {
+                ClearTransforms();
+
+                if (open.NewValue)
+                {
+                    AutoSizeAxes = Axes.Y;
+                    postsFlow.FadeIn(250, Easing.OutQuint);
+                }
+                else
+                {
+                    AutoSizeAxes = Axes.None;
+                    this.ResizeHeightTo(15, 250, Easing.OutQuint);
+
+                    postsFlow.FadeOut(250, Easing.OutQuint);
+                }
```

Comment:
You forgot about spacing between header and posts. Resizing posts to 0 will not remove spacing, instead there will be a small jump since spacing will be ignored after posts alpha will become 0, Using margin/padding for posts will not fix the issue. Using bottom margin for dropdown will not fix the issue as well since after posts becoming 0 height total dropdown height will be header height + header margin.

Reply:
You can add top margin for the posts fill flow container, and wrap it up in a container and perform all the autosizing and resizing logic on it, now the margin would be part of the autosizing.

Also you'll need `AlwaysPresent` on the container that has the autosizing workaround for `UpdateAfterAutoSize` to be invoked properly.

Reply:
Refactored in https://github.com/ppy/osu/pull/12736/commits/315a2b8314a044689eb607dfdced7bc24fc1f4c9